worker/caasprovisioner/k8s.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caasprovisioner
```

Comment:
I stopped here. While this works, and gets us closer to an MVP, I'm not comfortable pushing forward with this unless we split off into another prototype branch. Please separate the concerns of managing a connection to k8s from provisioning. Perhaps in two separate PRs, so it's clear to which worker/job the parts contribute.